drivers/sensors/ltr308.c

```diff
@@ -0,0 +1,1000 @@
+
+static int ltr308_set_reg8(FAR struct ltr308_dev_s *priv, uint8_t regaddr,
+                           uint8_t regval)
+{
+  struct i2c_msg_s msg;
+  uint8_t txbuffer[2];
+  int ret;
+
+  txbuffer[0]   = regaddr;
+  txbuffer[1]   = regval;
+
+  msg.frequency = priv->freq;
+  msg.addr      = priv->addr;
+  msg.flags     = 0;
+  msg.buffer    = txbuffer;
+  msg.length    = 2;
+
+  ret = I2C_TRANSFER(priv->i2c, &msg, 1);
+  if (ret < 0)
+    {
+      snerr("I2C_TRANSFER failed\n");
+    }
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: ltr308_get_reg8
+ *
+ * Description:
+ *   Read from an 8-bit LTR308 register
+ *
+ ****************************************************************************/
+
+static int ltr308_get_reg8(FAR struct ltr308_dev_s *priv, uint8_t regaddr,
+                           FAR uint8_t *regval)
+{
+  struct i2c_msg_s msg[2];
+  int ret;
+
+  msg[0].frequency = priv->freq;
+  msg[0].addr      = priv->addr;
+  msg[0].flags     = 0;
+  msg[0].buffer    = &regaddr;
+  msg[0].length    = 1;
+
+  msg[1].frequency = priv->freq;
+  msg[1].addr      = priv->addr;
+  msg[1].flags     = I2C_M_READ;
+  msg[1].buffer    = regval;
+  msg[1].length    = 1;
+
+  ret = I2C_TRANSFER(priv->i2c, msg, 2);
+  if (ret < 0)
+    {
+      snerr("I2C_TRANSFER failed: %d\n", ret);
+    }
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: ltr308_set_reg24
+ *
+ * Description:
+ *   Write to 3 8-bit LTR308 registers
+ *
+ ****************************************************************************/
+
+static int ltr308_set_reg24(FAR struct ltr308_dev_s *priv, uint8_t regaddr,
+                            uint32_t val)
+{
+  int i, ret;
+
+  for (i = 0; i < 3; i++, regaddr++)
+    {
+      ret = ltr308_set_reg8(priv, regaddr, val);
+      if (ret < 0)
+        {
+          return ret;
+        }
+
+      val >>= BYTE_TO_BITS;
+    }
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: ltr308_get_reg24
+ *
+ * Description:
+ *   Read from 3 8-bit LTR308 registers
+ *
+ ****************************************************************************/
+
+static int ltr308_get_reg24(FAR struct ltr308_dev_s *priv, uint8_t regaddr,
+                            FAR uint32_t *val)
+{
+  int i, ret;
+
+  *val = 0;
+  for (i = 0; i < 3; i++, regaddr++)
+    {
+      ret = ltr308_get_reg8(priv, regaddr, ((uint8_t *)val) + i);
+      if (ret < 0)
+        {
+          return ret;
+        }
+    }
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: ltr308_set_ctrl
+ *
+ * Description:
+ *   Power up, shut down or reset LTR308
+ *
+ ****************************************************************************/
+
+static inline int ltr308_set_ctrl(FAR struct ltr308_dev_s *priv,
```

Comment:
Maybe we can make a define instead

Reply:
If your concern is that the compiler might ignore the inline qualifier, we could replace it with the always_inline attribute. Would that be a good solution? Replacing it with a macro looks rather ugly to me..